src/components/manager/runs/RunLastContent.js

```diff
@@ -16,47 +17,116 @@ const COLUMNS_LAST_CONTENT = [
+            </Modal>
+        </>
+    );
+}
+
+FileNamesCell.propTypes = {
+    fileNames: PropTypes.arrayOf(PropTypes.string).isRequired,
+    dataType: PropTypes.string.isRequired,
+};
+
 const buildKeyFromRecord = (record) => `${record.dataType}-${record.tableId}`;
 
+const fileNameFromPath = (path) => path.split("/").pop();
+
+const getFileNameKey = (dataType, workflowParams) =>
+    dataType === "variant"
+        ? "vcf_gz.vcf_gz_file_names" // vcf_gz file names
+        : Object.keys(workflowParams)[0];
+
+const getDateFromEndTime = (endTime) => endTime.split("T")[0];
+
 const processIngestions = (data, currentTables) => {
-    const currentTableIds = (currentTables || []).map((table) => table.table_id);
+    const currentTableIds = new Set((currentTables || []).map((table) => table.table_id));
 
-    const ingestionsByDataType = data.reduce((acc, obj) => {
-        if (obj.state === "COMPLETE") {
-            const dataType = obj.details.request.tags.workflow_metadata.data_type;
-            const tableId = obj.details.request.tags.table_id;
+    const ingestionsByDataType = data.reduce((ingestions, run) => {
+        if (run.state === "COMPLETE") {
+            const dataType = run.details.request.tags.workflow_metadata.data_type;
+            const tableId = run.details.request.tags.table_id;
 
-            if (tableId === undefined || !currentTableIds.includes(tableId)) {
-                return acc;
+            if (tableId === undefined || !currentTableIds.has(tableId)) {
+                return ingestions;
             }
 
-            const fileNameKey = Object.keys(obj.details.request.workflow_params)[0];
-            const filePath = obj.details.request.workflow_params[fileNameKey];
-            const fileName = filePath.split("/").pop();
-            const dateStr = obj.details.run_log.end_time.split("T")[0];
-            const date = Date.parse(dateStr);
+            const fileNameKey = getFileNameKey(dataType, run.details.request.workflow_params);
+            const filePaths = run.details.request.workflow_params[fileNameKey];
+            const fileNames = Array.isArray(filePaths)
+                ? filePaths.map(fileNameFromPath)
+                : [fileNameFromPath(filePaths)];
 
-            const currentIngestion = { date: dateStr, dataType, tableId, fileNames: [fileName] };
+            const dateStr = getDateFromEndTime(run.details.run_log.end_time);
+            const date = Date.parse(dateStr);
 
+            const currentIngestion = { date: dateStr, dataType, tableId, fileNames };
             const dataTypeAndTableId = buildKeyFromRecord(currentIngestion);
 
-            if (!acc[dataTypeAndTableId] || date > Date.parse(acc[dataTypeAndTableId].date)) {
-                acc[dataTypeAndTableId] = currentIngestion;
-            } else if (date === Date.parse(acc[dataTypeAndTableId].date)) {
-                acc[dataTypeAndTableId].fileNames.push(fileName);
+            if (!ingestions[dataTypeAndTableId]) {
```

Comment:
Do we want to remove the date check for this `if` block?
If so, I would reverse the `if else` blocks this way to avoid `!`:
```javascript
if (ingestions[dataTypeAndTableId]) {
   // do stuff
} else { 
   // do different stuff
}
```